package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/lua
+local site = require 'gluon.site_config'
+local uci = require('luci.model.uci').cursor()
+
+if site.dns.servers
+then
+	dnsmasq=uci:get_first("dhcp", "dnsmasq")
+	uci:set('dhcp', dnsmasq, 'server', site.dns.servers)
+	uci:set('dhcp', dnsmasq, 'localise_queries', '1')
+	uci:set('dhcp', dnsmasq, 'localservice', '0')
+
+	if site.dns.cacheentries
+	then
+		uci:set('dhcp', dnsmasq, 'cachesize', site.dns.cacheentries)
+	end
+
+	if site.next_node.name
+	then
+		if site.next_node.ip4
+		then
+			uci:section('dhcp','domain','nextnode4',{
+				name=site.next_node.name,
+				ip=site.next_node.ip4,
+			}
+			)
+		end
+		if site.next_node.ip6
+		then
+			uci:section('dhcp','domain','nextnode6',{
+				name=site.next_node.name,
+				ip=site.next_node.ip6,
+			}
+			)
+		end
+	end
+	uci:save('dhcp')
+end
```

Comment:
You will still need to add these options to `check_site.lua` to validate the types of the values. There are many examples for optional values in other packages, marked as optional by passing *false* as last argument to the different check functions.